### PR TITLE
[api]Tracing sampling support

### PR DESCRIPTION
**Pull request by millken**
half of traces will be sampled
```
api:
  tracer:
    endpoint: http://localhost:14268/api/traces
    samplingRatio: .5
```

### CODE REVIEW

**Review by CoderZhi — APPROVED** [review]
the code of optionParams is hard to follow, but the function itself is working.

File: pkg/tracer/tracer.go [hunk 1]
```diff
@@ -76,11 +91,19 @@ func NewProvider(opts ...Option) (*tracesdk.TracerProvider, error) {
 	} else {
 		//skipped tracing when endpoint no set
 		return nil, nil
-		//trackerTracerProviderOption = append(trackerTracerProviderOption, tracesdk.WithSampler(tracesdk.ParentBased(tracesdk.TraceIDRatioBased(-1))))
 	}
 	if ops.serviceName == "" {
 		ops.serviceName = _service
 	}
+	if ops.samplingRatio != "" {
+		ratio, err := strconv.ParseFloat(ops.samplingRatio, 64)
```
Comment (Liuhaai) [inline pkg/tracer/tracer.go:99]: Do we need to judge the range of samplingratio here? (ratio < 0, ratio > 1)

Comment (millken) [reply pkg/tracer/tracer.go:99]: ratio >= 1 will always sample (default),< 0 are treated as zero will no sample

**Review by Liuhaai — APPROVED** [review]
